api/s2n.h

```diff
@@ -76,6 +76,11 @@ typedef enum {
 } s2n_max_frag_len;
 
 extern int s2n_config_add_cert_chain_and_key(struct s2n_config *config, const char *cert_chain_pem, const char *private_key_pem);
+extern int s2n_config_set_verification_ca_file(struct s2n_config *config, const char *ca_file_pem);
+typedef uint8_t (*s2n_verify_host_fn) (const char *host_name, size_t host_name_len, void *data);
```

Comment:
Would 'data' be more clear as 'ctx' for consistency?

Reply:
wrt your overarching concern, how to evaluate the host is something the application layer is best suited to decide. For example libcurl makes this decision not openssl or gnutls etc... I don’t think it’s a burden for the user. I think it’s a burden when the user doesn’t explicitly make the decision. This also helps with things like s3 virtual bucket adressing e.g my.bucket.amazonaws.com usually doesn’t work for our sdks and it results in a bad user experience. Letting the h2, http, or mqtt stack make this decision seems to be the best path IMHO.

I’m fine with using data universally. Will change